diff --git a/lib/Makefile.am b/lib/Makefile.am
index 902a7c8..573e055 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -70,13 +70,13 @@ ccnet-client.c: ccnet-object.h
 
 ccnet-object.h: ${ccnet_object_define}
 	rm -f $@
-	valac --pkg posix ${ccnet_object_define} -C -H ccnet-object.h
+	${VALAC} --pkg posix ${ccnet_object_define} -C -H ccnet-object.h
 
 ccnetobj.c: ${ccnet_object_define}
 	rm -f $@
-	valac -C --pkg posix ${ccnet_object_define}
+	${VALAC} -C --pkg posix ${ccnet_object_define}
 
-searpc_gen = searpc-signature.h searpc-marshal.h
+searpc_gen = searpc-marshal.h
 
 gensource: ${searpc_gen}
 
